controllers/component_build_controller_pac.go

```diff
@@ -500,12 +500,17 @@ func (r *ComponentBuildReconciler) generatePaCPipelineRunConfigs(ctx context.Con
 	if err != nil {
 		return nil, nil, err
 	}
+	pipelineName, pipelineBundle, err := getNameAndBundle(pipelineRef)
+	if err != nil {
+		// TODO: Log the error? Wrap the error and return it? Record an event?
+		return nil, nil, err
```

Comment:
How do we do error handling in build-service?

Reply:
'unsupported Tekton resolver " can be a transient error or normal Go error, that makes the reconciler be rescheduled to give a chance to correct the selector definition.

Refer to perror.go

Reply:
Would the user know that they have to correct the selector definition? Do the error messages for transient errors - or any other errors - somehow get propagated to the user?

Reply:
those errors are only in build-service pod logs

difference between transient and non transient (use of perror.go) is that controller either returns error or not
returns error : https://github.com/redhat-appstudio/build-service/blob/main/controllers/component_build_controller.go#L338
that will trigger another reconcile (and repeat few times)

if you don't return error like we do for permanent errors, it will proceed to the end and return nil
https://github.com/redhat-appstudio/build-service/blob/main/controllers/component_build_controller.go#L493
which won't run another reconcile

unsupported tekton resolver seems like persistent error to me

Reply:
It definitely should be a persistent error.

Reply:
Ack, added a permanent error